installer/bundle/bundle_skel_AIX.sh

```diff
@@ -0,0 +1,432 @@
+# Note that for non-Linux platforms, this symbol should contain full filename.
+#
+# PROVIDER_ONLY is normally set to '0'. Set to non-zero if you wish to build a
+# version of SCX that is only the provider (no OMI, no bundled packages). This
+# essentially provides a "scx-cimprov" type package if just the provider alone
+# must be included as part of some other package.
+
+TAR_FILE=<TAR_FILE>
+OM_PKG=<OM_PKG>
+OMI_PKG=<OMI_PKG>
+PROVIDER_ONLY=0
+
+SCRIPT_LEN=<SCRIPT_LEN>
+SCRIPT_LEN_PLUS_ONE=<SCRIPT_LEN+1>
+
+
+usage()
+{
+    echo "usage: $1 [OPTIONS]"
+    echo "Options:"
+    echo "  --extract              Extract contents and exit."
+    echo "  --force                Force upgrade (override version checks)."
+    echo "  --install              Install the package from the system."
+    echo "  --purge                Uninstall the package and remove all related data."
+    echo "  --remove               Uninstall the package from the system."
+    echo "  --restart-deps         Reconfigure and restart dependent service"
+    echo "  --source-references    Show source code reference hashes."
+    echo "  --upgrade              Upgrade the package in the system."
+    echo "  --version              Version of this shell bundle."
+    echo "  --debug                use shell debug mode."
+    echo "  -? | --help            shows this usage text."
+}
+
+source_references()
+{
+    cat <<EOF
+-- Source code references --
+EOF
+}
+
+cleanup_and_exit()
+{
+    # $1: Exit status
+    # $2: Non-blank (if we're not to delete bundles), otherwise empty
+
+    if [ -z "$2" -a -d "$EXTRACT_DIR" ]; then
+        cd $EXTRACT_DIR/..
+        rm -rf $EXTRACT_DIR
+    fi
+
+    if [ -n "$1" ]; then
+        exit $1
+    else
+        exit 0
+    fi
+}
+
+getVersionNumber()
+{
+    # Parse a version number from a string.
+    #
+    # Parameter 1: string to parse version number string from
+    #     (should contain something like mumble-4.2.2.135.universal.x86.tar)
+    # Parameter 2: prefix to remove ("mumble-" in above example)
+
+    if [ $# -ne 2 ]; then
+        echo "INTERNAL ERROR: Incorrect number of parameters passed to getVersionNumber" >&2
+        cleanup_and_exit 1
+    fi
+
+    echo $1 | sed -e "s/$2//" -e 's/\.universal\..*//' -e 's/\.x64.*//' -e 's/\.x86.*//' -e 's/-/./'
+}
+
+verifyNoInstallationOption()
+{
+    if [ -n "${installMode}" ]; then
+        echo "$0: Conflicting qualifiers, exiting" >&2
+        cleanup_and_exit 1
+    fi
+
+    return;
+}
+
+# $1 - The name of the package to check as to whether it's installed
+check_if_pkg_is_installed() {
+    lslpp $1 2> /dev/null 1> /dev/null
+    return $?
+}
+
+# $1 - The filename of the package to be installed
+# $2 - The package name of the package to be installed
+pkg_add() {
+    pkg_filename=$1
+    pkg_name=$2
+
+    echo "----- Installing package: $pkg_name ($pkg_filename) -----"
+    /usr/sbin/installp -a -X -d $pkg_filename ${pkg_name}.rte
```

Comment:
Wow, that's a whole lot simpler when it's not testing for 'n' different platforms!